Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesService.java

```diff
@@ -0,0 +1,389 @@
+            currPath = extFile;
+            try {
+                while (Environment.isExternalStorageRemovable(currPath.getParentFile())) {
+                    currPath = currPath.getParentFile();
+                }
+            } catch (IllegalArgumentException e) {
+                //swallow
+                return currPath.getAbsolutePath();
+            }
+        }
+        return currPath.getAbsolutePath();
+    }
+
+    private static List<File> searchDir(String mount) {
+        File photoDir = new File(mount + "/DCIM");
+        File[] goProDirs = photoDir.listFiles();
+        if (goProDirs == null || goProDirs.length == 0) {
+            return null;
+        }
+
+        List<File> photoFiles = new ArrayList<>();
+
+        for (File picDir : goProDirs) {
+            if (picDir.getName().toLowerCase().contains("gopro")) {
+                photoFiles.addAll(Arrays.asList(picDir.listFiles()));
+            }
+        }
+
+        return photoFiles;
+    }
+
+    private void sendFailedGeotaggingIntent(String failure) {
+        Intent intent = new Intent(STATE_FINISHED_GEOTAGGING);
+        intent.putExtra(EXTRA_SUCCESS, false);
+        intent.putExtra(EXTRA_FAILURE_MESSAGE, failure);
+        lbm.sendBroadcast(intent);
+    }
+
+    private class DownloadCameraTlogs extends AsyncTask<Object, Integer, Boolean> {
+
+        private WeakReference<Context> weakContext;
+
+        private DownloadCameraTlogs(Context context) {
+            this.weakContext = new WeakReference<>(context);
+        }
+
+        @Override
+        protected Boolean doInBackground(Object... params) {
+            Timber.d("Downloading the logs from Artoo");
+
+            try {
+
+                Context context = weakContext.get();
+                if (context != null) {
+                    File folder = context.getExternalFilesDir(null);
+                    if (folder == null) {
+                        return false;
+                    }
+
+                    final boolean isSoloDownloadSuccessful = downloadFileFromDevice
+                        (CAMERA_TLOG_FILE, folder, soloSshLink, NUMBER_OF_RETRIES);
+
+                    return isSoloDownloadSuccessful;
+
+                }
+            } catch (IOException e) {
+                Timber.e("Unable to download the file to the phone.", e);
+            } catch (NullPointerException e) {
+                Timber.e("Unable to create file path. ", e);
+            }
+            return false;
+        }
+
+        protected void onProgressUpdate(Integer... progress) {
+            super.onProgressUpdate(progress);
+        }
+
+        /**
+         * After completing background task Dismiss the progress dialog
+         */
+        protected void onPostExecute(Boolean result) {
+            Intent intent = new Intent(STATE_FINISHED_LOADING_LOGS);
+            intent.putExtra(EXTRA_SUCCESS, result);
+            lbm.sendBroadcast(intent);
+        }
```

Comment:
@chaviw nevermind. I looked at the javadocs comment without looking at the code.